test_conformance/images/kernel_read_write/test_common.h

```diff
@@ -63,8 +59,10 @@ int determine_validation_error_offset(
     bool clampingErr = false, clamped = false, otherClampingBug = false;
     int clampedX, clampedY, clampedZ;
 
-    size_t imageWidth = imageInfo->width, imageHeight = imageInfo->height,
-           imageDepth = imageInfo->depth;
+    size_t imageWidth, imageHeight, imageDepth;
+    cl_int error =
+        get_image_dimensions(imageInfo, imageWidth, imageHeight, imageDepth);
+    test_error(error, "invalid image dimensions");
```

Comment:
If _get_image_dimensions_ returns a _test_status_, we shouldn't be passing that status to _test_error_.

Reply:
Changed this around a bit, does it looks better?
Quick note: This is the first of many more pull requests to refactor this code, unless there are more issues specific to this patch (get_image_dimensions will probably stay as is) can we revisit this on a later change once the code has settled into its final shape?

Reply:
Looks better.  I agree about dealing with the remaining small details later.